app/Http/Controllers/LA/ConfigureMACDBotsController.php

```diff
@@ -369,4 +369,100 @@ public function getChart($symbol) {
         }
     }
 
+    /**
+     * Get the MACD Chart based in user selection and MACD settings
+     * 
+     * @param type $symbol
+     * @return type
+     */
+    public function getChart($symbol) {
+        $api = new \Binance\API($this->APIKey, $this->ScreateKey);
+
+        /**
+         * https://cryptobee.itbutton.com/data_import/binance/IOTABTC 
+         * http://jsonviewer.stack.hu/#http://cryptobee.itbutton.com/data_import/binance/IOTABTC
+         *
+         ***/
+        $myRecentData = $api->candlesticks(strtoupper($symbol), "3m");
+        $totalRecentData = count($myRecentData);
+        
+        if ($totalRecentData > 0) {
+            $ema_short_period = 12;
+            $ema_long_period = 26;
+            $signal_period = 9;
+            
+            if( Auth::check()) {
+                $symbols = DB::table('configuremacdbots')->select("*")
+                        ->where('userid', '=', Auth::user()->id)
+                        ->where('symbol', '=', $symbol)
+                        ->whereNull('deleted_at')
+                        ->get();
+                if(count($symbols) > 0) {
+                    $ema_short_period = $symbols[0]->ema_short_period;
+                    $ema_long_period = $symbols[0]->ema_long_period;
+                    $signal_period = $symbols[0]->signal_period;
+                }
+            }
+            
+            
+            array_walk($myRecentData, function(&$item, $key) {
+                $timeZoneOffSet = 3.30 * 3600;
+                $item["closeTimeL"] = $item["closeTime"];
+                $item["openTimeL"] = $item["openTime"];
+                $item["closeTime"] = date("Y-m-d H:i:s", (($item["closeTime"] + $timeZoneOffSet) / 1000));
+                $item["openTime"] = date("Y-m-d H:i:s", (($item["openTime"] + $timeZoneOffSet) / 1000));
+                $item["keyTime"] = date("Y-m-d", (($key + $timeZoneOffSet) / 1000));
+                $item["timestamp"] = ($key);
+            });
+
+            $closeArray = array_values(array_map(function($sub) {
+                        return $sub['close'];
+                    }, $myRecentData));
+                    
+            $macdData = [];
+            $ema_fast = Average::exponentialMovingAverage($closeArray, $ema_short_period); /** 12 **/
+            $ema_slow = Average::exponentialMovingAverage($closeArray, $ema_long_period); /** 26 **/
+            $macd = $signal = $hist = array();
+            
+            $macd = $this->subtractTwoArray($ema_fast, $ema_slow);
+            $signal = Average::exponentialMovingAverage($macd, $signal_period);       /** 9 **/
+            $hist = $this->subtractTwoArray($macd, $signal);
+            
+            $keys = array_keys($myRecentData);
+            
+            for($i = 0; $i < $totalRecentData; $i++) {
```

Comment:
I think the API returns the oldest data first, so now you pick the oldest data first. You should pick the latest data from the API response.

Reply:
Yes, That we need to make changes.